nilearn/tests/test_signal.py

```diff
@@ -627,21 +697,19 @@ def test_clean_confounds_errros(signals):
     with pytest.raises(TypeError, match="confound has an unhandled type"):
         clean(signals, confounds=[None])
 
-    with pytest.raises(
-        ValueError, match="Confound signal has an incorrect length"
-    ):
+    msg = "Confound signal has an incorrect length."
+    with pytest.raises(ValueError, match=msg):
         clean(signals, confounds=np.zeros(2))
-    with pytest.raises(
-        ValueError, match="Confound signal has an incorrect length"
-    ):
+    with pytest.raises(ValueError, match=msg):
         clean(signals, confounds=np.zeros((2, 2)))
-    with pytest.raises(
-        ValueError, match="Confound signal has an incorrect length."
-    ):
+    with pytest.raises(ValueError, match=msg):
         current_dir = Path(__file__).parent
         filename1 = current_dir / "data" / "spm_confounds.txt"
         clean(signals[:-1, :], confounds=filename1)
 
+
+def test_clean_errros(signals):
+    """Test error handling."""
```

Comment:
This could also be parameterized.

Reply:
same here about the readability of the test.